src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoException.scala

```diff
@@ -12,3 +14,7 @@ abstract class LeoException(
     ErrorReport(Option(getMessage).getOrElse(""), Some(statusCode), Seq(), Seq(), Some(this.getClass))
   }
 }
+
+final case class RequestValidationError(message: String) extends NoStackTrace {
```

Comment:
Should this extend `LeoException`?

Reply:
I didn't really want stack trace since it won't be very useful in this case (the message will be unique enough) and stack trace is the expensive part when creating exceptions.

Reply:
oh I just realize there's no stack trace in leoException.

Reply:
Yeah, LeoException doesn't send a stack trace but it defines a status code and a message. Slight preference for standardizing on `LeoException` for API errors.

Reply:
Talked to @rtitle offline. The idea is I want to distinct internal exceptions vs error reporting ones. I'll leave this PR as is and take another look at usage of exceptions later, and potentially remove `this.getClass` from errors we report to user.